src/trio/_core/_run.py

```diff
@@ -543,16 +543,60 @@ class CancelScope:
 
     # Constructor arguments:
     _deadline: float = attrs.field(default=inf, kw_only=True, alias="deadline")
+    # use float|None, or math.inf?
+    # or take it as an opportunity to also introduce deadline=None?
+    _relative_deadline: float | None = attrs.field(
+        default=None, kw_only=True, alias="relative_deadline"
+    )
     _shield: bool = attrs.field(default=False, kw_only=True, alias="shield")
 
+    # introduced during deprecation period of handling relative timeouts from
+    # initialization
+    _timeout_from_enter: bool = attrs.field(
+        default=True, kw_only=True, alias="_timeout_from_enter"
+    )
+    _creation_time: float | None = attrs.field(default=None, alias="_creation_time")
+
     @enable_ki_protection
     def __enter__(self) -> Self:
         task = _core.current_task()
         if self._has_been_entered:
             raise RuntimeError(
                 "Each CancelScope may only be used for a single 'with' block"
             )
+
+        if (
+            self._creation_time is not None
+            and abs(self._creation_time - current_time()) > 0.01
+            and not self._timeout_from_enter
+        ):
+            # not using warn_deprecated because the message template is a weird fit
+            # TODO: mention versions in the message?
```

Comment:
I think the trio deprecation util does this

Reply:
possibly, or I could add some kwarg that overrides wording. But since not using `TrioDeprecationWarning` I don't really feel like `warn_deprecated` adds much of anything in a lot of cases
https://github.com/python-trio/trio/blob/a7db0e43ed9f5bbccabed7a06bf3717e3f8d093e/src/trio/_deprecate.py#L54-L75